communications/telegram/telegram.go

```diff
@@ -0,0 +1,230 @@
+
+	talkRoot = "GoCryptoTrader bot"
+)
+
+// Telegram is the overarching type across this package
+type Telegram struct {
+	base.Base
+	Token             string
+	Offset            int64
+	AuthorisedClients []int64
+}
+
+// Setup takes in a Telegram configuration and sets verification token
+func (t *Telegram) Setup(config config.CommunicationsConfig) {
+	t.Name = config.TelegramConfig.Name
+	t.Enabled = config.TelegramConfig.Enabled
+	t.Token = config.TelegramConfig.VerificationToken
+	t.Verbose = config.TelegramConfig.Verbose
+}
+
+// Connect starts an initial connection
+func (t *Telegram) Connect() error {
+	if err := t.TestConnection(); err != nil {
+		return err
+	}
+	t.Connected = true
+	go t.PollerStart()
+	return nil
+}
+
+// PushEvent sends an event to a supplied recipient list via telegram
+func (t *Telegram) PushEvent(event base.Event) error {
+	for i := range t.AuthorisedClients {
+		err := t.SendMessage(fmt.Sprintf("Type: %s Details: %s GainOrLoss: %s",
+			event.Type, event.TradeDetails, event.GainLoss), t.AuthorisedClients[i])
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// PollerStart starts the long polling sequence
+func (t *Telegram) PollerStart() {
+	t.InitialConnect()
+
+	for {
+		resp, err := t.GetUpdates()
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		for i := range resp.Result {
+			if resp.Result[i].UpdateID > t.Offset {
+				if string(resp.Result[i].Message.Text[0]) == "/" {
+					err = t.HandleMessages(resp.Result[i].Message.Text, resp.Result[i].Message.From.ID)
+					if err != nil {
+						log.Fatal(err)
+					}
+				}
+				t.Offset = resp.Result[i].UpdateID
+			}
+		}
+	}
+}
+
+// InitialConnect sets offset, and sends a welcome greeting to any associated
+// IDs
+func (t *Telegram) InitialConnect() {
+	resp, err := t.GetUpdates()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if !resp.Ok {
+		log.Fatal(resp.Description)
+	}
+
+	warmWelcomeList := make(map[string]int64)
+	for i := range resp.Result {
+		if resp.Result[i].Message.From.ID != 0 {
+			warmWelcomeList[resp.Result[i].Message.From.UserName] = resp.Result[i].Message.From.ID
+		}
+	}
+
+	for userName, ID := range warmWelcomeList {
+		err = t.SendMessage(fmt.Sprintf("GoCryptoTrader bot has connected: Hello, %s!", userName), ID)
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+	if len(resp.Result) == 0 {
+		return
+	}
+	t.Offset = resp.Result[len(resp.Result)-1].UpdateID
+}
+
+// HandleMessages handles incoming message from the long polling routine
+func (t *Telegram) HandleMessages(text string, chatID int64) error {
+	switch {
+	case common.StringContains(text, cmdHelp):
+		return t.SendMessage(fmt.Sprintf("%s: %s", talkRoot, cmdHelpReply), chatID)
+
+	case common.StringContains(text, cmdStart):
+		return t.SendMessage(fmt.Sprintf("%s: START COMMANDS HERE", talkRoot), chatID)
+
+	case common.StringContains(text, cmdOrders):
+		return t.SendMessage(fmt.Sprintf("%s: %s", talkRoot, t.GetOrderbook("ANX")), chatID)
```

Comment:
Need to check to see if the supplied exchange is valid, and if so, get the orderbook for the desired exchange. Will also need to take in a param for a currency pair.